components/blocks/FilesToCards.vue

```diff
@@ -59,26 +61,229 @@
+                :href="item.links.documentation"
+                ><span
+                  >DOCUMENTATION<span class="icon ml-2"
+                    ><i class="mdi mdi-book" /></span></span
+              ></a>
+              <a
+                v-if="item.links.publication"
+                class="link-item d-button has-text-link"
+                :href="item.links.publication"
+                ><span
+                  >PUBLICATION<span class="icon ml-2"
+                    ><i class="mdi mdi-newspaper" /></span></span
+              ></a>
+              <a
+                v-if="item.links.doi"
+                class="link-item d-button has-text-link"
+                :href="'https://' + item.links.doi"
+                ><span
+                  >PUBLICATION<span class="icon ml-2"
+                    ><i class="mdi mdi-newspaper" /></span></span
+              ></a>
+            </div>
+          </template>
+        </DCard>
+      </div>
+    </div>
   </main>
 </template>
 
 <script>
 import DCard from '@/components/base/DCard.vue';
+import Multiselect from 'vue-multiselect';
 
 export default {
   components: {
     DCard,
+    Multiselect,
   },
   props: {
     data: {
       type: Array,
       required: true,
     },
   },
+  data: () => ({
+    ascending: true,
+    sortBy: [],
+    searchGroup: [],
+    sortOptions: [],
+  }),
   computed: {
     filteredData() {
       return this.data.filter((d) => !d.hidden);
     },
+    sortedArray() {
+      let tempCards = this.data;
+
+      // Filter by group tag
+      let filtered = tempCards;
+      if (this.searchGroup.length > 0) {
+        filtered = [];
+        for (const arr in tempCards) {
+          for (const filter in this.searchGroup) {
+            if (tempCards[arr].group === this.searchGroup[filter]) {
+              filtered.push(tempCards[arr]);
+            }
+          }
+        }
+      }
+
+      // Sort by alphabetical order
+      if (this.sortBy.name === 'Title') {
+        tempCards = filtered.sort((a, b) => {
+          const fa = a.title.toLowerCase();
+          const fb = b.title.toLowerCase();
+          if (fa < fb) {
+            return -1;
+          }
+          if (fa > fb) {
+            return 1;
+          }
+          return 0;
+        });
+      } else if (this.sortBy.name === 'Date') {
+        // Sort by date
+        tempCards = filtered.sort((a, b) => {
+          return new Date(a.date) - new Date(b.date);
+        });
+      }
+
+      // Show sorted array in descending or ascending order
+      if (!this.ascending) {
+        filtered.reverse();
+      }
+
+      return filtered;
+    },
+  },
+  methods: {
+    cardTags() {
+      const tags = [];
+      this.data.forEach(function (obj) {
+        tags.push(obj.group);
+      });
+      const uniqueTags = tags.filter(function (item, pos) {
+        return tags.indexOf(item) === pos;
+      });
+      return uniqueTags.sort();
+    },
+    sortByOptions() {
+      let condition = false;
```

Comment:
I'd change this to:
```js
const hasDate = this.data.some((card) => card.hasOwnProperty("date"))

const options = [{ name: 'Title' }]
if (hasDate) options.push({ name: 'Date'});

return options;
```

Reply:
I get error:  Do not access Object.prototype method 'hasOwnProperty' from target object

Reply:
Is that a linting error? or a runtime error?

Reply:
[eslint](https://eslint.org/docs/rules/no-prototype-builtins) says to do `Object.prototype.hasOwnProperty.call(card, "date")` instead

Reply:
Or we can just ignore it since this is a hardcoded key and not actually a security risk by adding `// eslint-disable-next-line no-prototype-builtins` the line above